### PR TITLE
- Add option to add you own loading to fix Problem of Loader doesn't …

### PR DESCRIPTION
**Pull request by hossamCheck24**
- Add option to add you own loading to fix Problem of Loader doesn't fill the screen

### CODE REVIEW

File: xs2awizard/src/main/java/com/fintecsystems/xs2awizard/components/XS2AWizardViewModel.kt [hunk 1]
```diff
@@ -111,6 +111,7 @@ class XS2AWizardViewModel(
     fun onStop() {
         // Cleanup in case the viewModel gets reused for a future request.
         loadingIndicatorLock.value = false
+        config.onLoading.invoke(false)
```
Comment (maik-mursall) [inline xs2awizard/src/main/java/com/fintecsystems/xs2awizard/components/XS2AWizardViewModel.kt:114]:
```suggestion
        config.onLoading(false)
```

Comment (maik-mursall) [reply xs2awizard/src/main/java/com/fintecsystems/xs2awizard/components/XS2AWizardViewModel.kt:114]: It's possible to directly call the method

File: xs2awizard/src/main/java/com/fintecsystems/xs2awizard/components/XS2AWizardViewModel.kt [hunk 2]
```diff
@@ -231,6 +232,7 @@ class XS2AWizardViewModel(
     private fun submitForm(jsonBody: String, showIndicator: Boolean) {
         if (showIndicator) {
             loadingIndicatorLock.value = true
+            config.onLoading.invoke(true)
```
Comment (maik-mursall) [inline xs2awizard/src/main/java/com/fintecsystems/xs2awizard/components/XS2AWizardViewModel.kt:235]:
```suggestion
            config.onLoading(true)
```

File: xs2awizard/src/main/java/com/fintecsystems/xs2awizard/components/XS2AWizardViewModel.kt [hunk 3]
```diff
@@ -319,6 +321,7 @@ class XS2AWizardViewModel(
         }
 
         loadingIndicatorLock.value = false
+        config.onLoading.invoke(false)
```
Comment (maik-mursall) [inline xs2awizard/src/main/java/com/fintecsystems/xs2awizard/components/XS2AWizardViewModel.kt:324]:
```suggestion
        config.onLoading(false)
```

**Comment by maik-mursall** [comment]
Hi @hossamCheck24,
thank you again for the PR :)

Adding a `onLoading` callback seems fair enough, but I think it may also be good to have a way to disable the default loading indicator.
Maybe adding another field to the `XS2AWizardConfig` like `enableDefaultLoadingIndicator` and setting its default value to `true`?
At least I think that should be present, so you will be able to entirely use your own loading indicator implementation.

Please let me know what you think about this :)

**Comment by hossamCheck24** [comment]
Hi @maik-mursall 
Thanks for code review :D 
i will work on it  and will let you know when it finished 
thanks a lot

**Pull request closed without merging**